sections/monte_carlo/quantum_monte_carlo/quantum_monte_carlo.tex

```diff
@@ -97,71 +311,259 @@ \subsubsection{Construct the payoff function}\label{sec:MC_Payoff}
+                  else:
+                      comparator.ccx(qubits_dist[i], qubits_compare[i], qubits_compare[i + 1])
+              else:
+                  if bin_compare_value[i] == 1:
+                      # OR needs the result argument as qubit not register, thus
+                      # access the index [0]
+                      comparator.compose(OR(2), [qubits_dist[i], qubits_compare[i], qubits_compare[0]], inplace=True)
+                  else:
+                      comparator.ccx(qubits_dist[i], qubits_compare[i], qubits_compare[0])
+          # flip result bit if geq flag is false
+          if not bigger_then:
+              comparator.x(qubits_compare[0])
+          # uncompute ancillas state
+          for i in reversed(range(num_qubits_dist - 1)):
+              if i == 0:
+                  if bin_compare_value[i] == 1:
+                      comparator.cx(qubits_dist[i], qubits_compare[i+1])
+              else:
+                  if bin_compare_value[i] == 1:
+                      comparator.compose(OR(2), [qubits_dist[i], qubits_compare[i], qubits_compare[i+1]], inplace=True)
+                  else:
+                      comparator.ccx(qubits_dist[i], qubits_compare[i], qubits_compare[i+1])
+      else:
+          # num_state_qubits == 1 and value == 1:
+          comparator.cx(qubits_dist[0], qubits_compare[0])
+          # flip result bit if geq flag is false
+          if not bigger_then:
+              comparator.x(qubits_compare[0])
+  else:
+      if not bigger_then:  # otherwise the condition is never satisfied
+          comparator.x(qubits_compare[0])
+  comparator.name = name.format(value=compare_value, bit_value="".join(str(x) for x in bin_compare_value))
+  return comparator
+\end{lstlisting}
 
-Now that we have shown how to construct a piece-wise linear function in a quantum computer we change $\theta_P$ again to better obtain our desired result $\mathbb{E}$.
+\subsubsection{Implementation of piece-wise linear functions}\label{sec:ImplPieceFunc}
+
+Implementing the piecewise linear function is not directly straigth forward so a detailed introduction is given in the following. 
+As mentioned above we are dealing with linear function to encode the payoff
+\begin{equation}\label{eq:defLinearFunc}
+  f(x) = \alpha x + \beta,
+\end{equation}
+where $f(x)$ maps $x$ to the payoff by equation \ref{eq:callPayoff} and the domain of $x$ consist of of two linear function. One for $S_T < K$ and one for $S_T \geq K$. 
+Furthermore the $S_i$ do not occure as a real number but as an element of $\{ 0,...,2^n-1 \}$ and additionally on a quantum computer $f(x)$ has to be realised
+using rotations in the corresponding Hilbert space. 
+These challenges require us to use a affine mapping $\hat{f}(x)$ which is in contrast to
+\begin{equation}\label{eq:linFuncBound}
+  f: [ a,b ] \rightarrow [ c,d ]
+\end{equation}
+defined as
+\begin{equation}\label{eq:hatLinFunc}
+  \hat{f}: \{0,...,2^n-1 \} \rightarrow [ 0,1 ].
```

Comment:
I think you want the function \hat{f} to be symmetric around 0 for the taylor approximation of sin^2(.) to make sense. Therefore the range should be from [-1,1]